File: data-ingestion/models/company_overview.py
from dataclasses import dataclass
from datetime import date
from typing import Optional

@dataclass
class CompanyOverview:
	symbol: str
	asset_type: str
	name: str
	description: str
	cik: str
	exchange: str
	currency: str
	country: str
	sector: str
	industry: str
	address: str
	fiscal_year_end: str
	latest_quarter: date
	market_capitalization: int
	ebitda: int
	pe_ratio: float
	peg_ratio: float
	book_value: float
	dividend_per_share: Optional[float]
	dividend_yield: float
	eps: float
	revenue_per_share_ttm: float
	profit_margin: float
	operating_margin_ttm: float
	return_on_assets_ttm: float
	return_on_equity_ttm: float
	revenue_ttm: int
	gross_profit_ttm: int
	diluted_eps_ttm: float
	quarterly_earnings_growth_yoy: float
	quarterly_revenue_growth_yoy: float
	analyst_target_price: float
	trailing_pe: float
	forward_pe: float
	price_to_sales_ratio_ttm: float
	price_to_book_ratio: float
	ev_to_revenue: float
	ev_to_ebitda: float
	beta: float
	high_52_week: float
	low_52_week: float
	moving_average_50_day: float
	moving_average_200_day: float
	shares_outstanding: int
	dividend_date: Optional[date]
	ex_dividend_date: Optional[date]

File: StockScreenerWebApp/Models/StockScreenerDb/CompanyOverview.cs
namespace StockScreenerWebApp.Models.StockScreenerDb
{
    public partial class CompanyOverview
    {
        public string Symbol { get; set; } = null!;
        public string AssetType { get; set; } = null!;
        public string Name { get; set; } = null!;
        public string Description { get; set; } = null!;
        public string Cik { get; set; } = null!;
        public string Exchange { get; set; } = null!;
        public string Currency { get; set; } = null!;
        public string Country { get; set; } = null!;
        public string Sector { get; set; } = null!;
        public string Industry { get; set; } = null!;
        public string Address { get; set; } = null!;
        public string FiscalYearEnd { get; set; } = null!;
        public DateTime LatestQuarter { get; set; }
        public long MarketCapitalization { get; set; }
        public long? Ebitda { get; set; }
        public double? Peratio { get; set; }
        public double? Pegratio { get; set; }
        public double? BookValue { get; set; }
        public double? DividendPerShare { get; set; }
        public double? DividendYield { get; set; }
        public double? Eps { get; set; }
        public double? RevenuePerShareTtm { get; set; }
        public double? ProfitMargin { get; set; }
        public double? OperatingMarginTtm { get; set; }
        public double? ReturnOnAssetsTtm { get; set; }
        public double? ReturnOnEquityTtm { get; set; }
        public long? RevenueTtm { get; set; }
        public long? GrossProfitTtm { get; set; }
        public double? DilutedEpsttm { get; set; }
        public double? QuarterlyEarningsGrowthYoy { get; set; }
        public double? QuarterlyRevenueGrowthYoy { get; set; }
        public double? AnalystTargetPrice { get; set; }
        public double? TrailingPe { get; set; }
        public double? ForwardPe { get; set; }
        public double? PriceToSalesRatioTtm { get; set; }
        public double? PriceToBookRatio { get; set; }
        public double? EvtoRevenue { get; set; }
        public double? EvtoEbitda { get; set; }
        public double? Beta { get; set; }
        public double? _52weekHigh { get; set; }
        public double? _52weekLow { get; set; }
        public double? _50dayMovingAverage { get; set; }
        public double? _200dayMovingAverage { get; set; }
        public long? SharesOutstanding { get; set; }
        public DateTime? DividendDate { get; set; }
        public DateTime? ExDividendDate { get; set; }

        public virtual Symbol SymbolNavigation { get; set; } = null!;
    }
}
